Library/Homebrew/keg_relocate.rb

```diff
@@ -86,17 +86,24 @@ def change_rpath(file, new_prefix)
       return
     end
     return unless patchelf.installed?
-    cmd = "#{patchelf.bin}/patchelf --set-rpath #{new_prefix}/lib"
-    (old_i, old_p) = `#{patchelf.bin}/patchelf --print-rpath --print-interpreter #{file}`.split("\n")
+    old_path = `#{patchelf.bin}/patchelf --print-rpath #{file}`.strip
+    old_interp = ""
     raise ErrorDuringExecution.new(cmd) unless $?.success?
+    lib_path = "#{new_prefix}/lib"
+    rpath = old_path.split(":").map { |x| x.sub(old_prefix, new_prefix) }
+    rpath << lib_path unless rpath.include? lib_path
+    new_rpath = rpath.join(":")
+    cmd = "#{patchelf.bin}/patchelf --set-rpath #{new_rpath}"
     if file.mach_o_executable?
+      old_interp = `#{patchelf.bin}/patchelf --print-interpreter #{file}`.strip
```

Comment:
Can you split this PR into two commits, one two fix the `cannot find section .interp` bug and one to relocate `RPATH`?